src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,243 @@
+    * @param joint2 Second input joint
+    * @param inputBlock Block that is to be subdivided
+    * @param desiredVolume Desired volume of subdivision(s)
+    * @param totalVolume Total volume of rock volume in analysis
+    * @param tolerance Acceptable deviation from ideal volume per processor
+    * @return Returns tuple containing joint that provides best division of volume and remaining volume that is to be
+    *         subdivided further. If neither joint yields acceptable subdivision, function returns None.
+    */
+  private def testVolumes(joint1: Joint, joint2: Joint, inputBlock: Block,
+                          desiredVolume: Double, totalVolume: Double,
+                          tolerance: Double): Option[(Joint, Block)] = {
+    val blockSet1 = inputBlock cut joint1
+    val blockSet2 = inputBlock cut joint2
+
+    val nonRedundantBlocks = blockSet1 map { case block @ Block(blockCenter, _, _) =>
+        Block(blockCenter, block.nonRedundantFaces)
+    }
+    val nonRedundantBlocks2 = blockSet2 map { case block @ Block(blockCenter, _, _) =>
+      Block(blockCenter, block.nonRedundantFaces)
+    }
+
+    val sortedBlocks1 = nonRedundantBlocks.sortWith(_.volume < _.volume)
+    val sortedBlocks2 = nonRedundantBlocks2.sortWith(_.volume < _.volume)
+
+    if (sortedBlocks1.length == 1) {
+      // joint1 does not intersect the block
+      if (sortedBlocks2.length == 1) {
+        // Neither joint intersects the block
+        None
+      } else if (sortedBlocks2(0).volume >= desiredVolume) {
+        // joint2 sufficiently divides block, remaining volume should still be further subdivided
+        Some(joint2, sortedBlocks2(1))
+      } else if (sortedBlocks2(1).volume <= desiredVolume) {
+        // joint2 sufficiently divides block, remaining volume requires no further subdivision
+        Some(joint2, sortedBlocks2(1))
+      } else {
+        // joint 2 does not divide block adequately
+        None
+      }
+    } else if (sortedBlocks2.length == 1) {
+      // joint 2 does not intersect the block
+      if (sortedBlocks1(0).volume >= desiredVolume) {
+        // joint1 sufficiently divides block, remaining volume should still be further subdivided
+        Some(joint1, sortedBlocks1(1))
+      } else if (sortedBlocks1(1).volume <= desiredVolume) {
+        // joint1 sufficiently divides block, remaining volume requires no further subdivision
+        Some(joint1, sortedBlocks1(1))
+      } else {
+        // joint 1 does not divide the block adequately
+        None
+      }
+    } else {
+      // BOTH joints intersect the block
+      // Compare volumes to desired volumes
+      val volumeDiff1 = (sortedBlocks1(0).volume - desiredVolume) / desiredVolume
+      val volumeDiff2 = (sortedBlocks2(0).volume - desiredVolume) / desiredVolume
+
+      // Return joint that gives volume closest to desired volume
+      // When less than 50% of the total volume remains, begin loosening constaints on joint selected until
```

Comment:
What determined the choice of 50% as the threshold? Is it just arbitrary? If so, we may want to extract this out as a constant.